### PR TITLE
[BTFS-974] - gRPC panic middleware/interceptors common

### PR DESCRIPTION
**Pull request by uchetron**
1. Added 	a panic handler that prints the stack trace when recovering from a panic.

2. Added shared options for the logger, with a custom gRPC code to log level function.

3. Added UnaryServerInterceptor handle.

(implementation found in status-server - https://github.com/TRON-US/status-server/pull/74)

### CODE REVIEW

File: handlers/grpcpanichandler.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package handlers
+
+import (
+	"errors"
+	"runtime"
+
+	"github.com/tron-us/go-common/log"
```
Comment (taiyangc) [inline handlers/grpcpanichandler.go]: Use go-common/v2

Comment (uchetron) [reply handlers/grpcpanichandler.go]: ok

Comment (uchetron) [reply handlers/grpcpanichandler.go]: okay.

File: handlers/grpcpanichandler.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package handlers
```
Comment (wxue) [inline handlers/grpcpanichandler.go]: should be middleware package, handlers usually at service level

File: go.mod [hunk 3]
```diff
@@ -6,13 +6,10 @@ require (
 	github.com/btcsuite/btcutil v0.0.0-20190425235716-9e5f4b9a998d
 	github.com/go-pg/migrations v6.7.3+incompatible
 	github.com/go-pg/pg v0.0.0-20190627115636-374b7dab11ff
-	github.com/jinzhu/inflection v1.0.0 // indirect
+	github.com/grpc-ecosystem/go-grpc-middleware v1.1.0
```
Comment (wxue) [inline go.mod]: plz make sure rebase master and tidy before merge

Comment (uchetron) [reply go.mod]: ok

File: handlers/grpcpanichandler.go (hunk 2 above)
Comment (uchetron) [reply handlers/grpcpanichandler.go]: changed to middleware.

File: middleware/grpcmiddleware.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	grpc_recovery "github.com/grpc-ecosystem/go-grpc-middleware/recovery"
+)
+
+var (
+	// Panic handler prints the stack trace when recovering from a panic.
+	RecoveryCustomFunc grpc_recovery.RecoveryHandlerFunc = grpc_recovery.RecoveryHandlerFunc(func(p interface{}) error {
+		buf := make([]byte, 1<<16)
```
Comment (wxue) [inline middleware/grpcmiddleware.go]: for your consideration: https://golang.org/pkg/bytes/#Buffer
if it can help to make the code cleaner

Comment (uchetron) [reply middleware/grpcmiddleware.go]: ok.

**Comment by uchetron** [comment]
@jin-tron @taiyangc @JustinGaoF please review

**Pull request closed without merging**